### PR TITLE
Generate Assertion Ops for interpolate_bilinear

### PR DESCRIPTION
**Pull request by faizan-m**
# Description

Enables the assertions to be included as part of the traced graph allowing functions traced with shapeless Tensors to be used for shaped instances and throw the proper exceptions.

Removed `test_symbolic_tensor_shape` because it is not supported as per KerasTensor update.

This CL prevents the updated code from being broken on the rollout of enabling subtyping for tf.function.

Fixes # (issue)

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [x] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

**Comment by bhack** [comment]
Many SIG members have no visibility to the internal  CL

**Comment by faizan-m** [comment]
Hi bhack! 

Sorry about that! I have added more details!

**Comment by bhack** [comment]
> Enables the assertions to be included as part of the traced graph allowing functions traced with shapeless Tensors to be used for shaped instances and throw the proper exceptions.

I understand this but do we have this documented somewhere?

**Comment by faizan-m** [comment]
I am confused regarding what documentation would be ideal here?

To put another way, there is currently a bug in the code. If you have a generalized function:
fn = interpolate_bilinear.get_concrete_function(
      tf.TensorSpec(shape=None, dtype=tf.float32),
      tf.TensorSpec(shape=None, dtype=tf.float32),
  )
And you call it on an invalid shape, say [1, 2, 1, 1]. Then error you get is not "Grid width must be at least 2." but "indices[0,0] = -1 is not in [0, 2)" that is being generated at a much lower level and is difficult to understand.

**Comment by bhack** [comment]
Ok, we have some failing test.

> tensorflow.python.framework.errors_impl.InvalidArgumentError:  indices[0,0] = -1 is not in [0, 2)

**Comment by faizan-m** [comment]
Yeah those probably come from:  https://github.com/tensorflow/addons/issues/138

I think tf.function is stochastically parallelizing the asserts (which it shouldn't) leading to the flakiness.

I fixed them by using explicit control dependencies but it appears that the code compliance check is not happy with that. Even if the assert bug is fixed at head, the flakiness will persist for previous TF versions. So I think for the sake of code functionality, we should consider adding this file to the exceptions list.

File: tensorflow_addons/image/dense_image_warp.py [hunk 1]
```diff
@@ -46,33 +46,56 @@ def interpolate_bilinear(
...
-        grid = tf.convert_to_tensor(grid)
-        query_points = tf.convert_to_tensor(query_points)
+    grid = tf.convert_to_tensor(grid)
+    query_points = tf.convert_to_tensor(query_points)
+    grid_shape = tf.shape(grid)
+    query_shape = tf.shape(query_points)
+
+    with tf.control_dependencies(
+        [
+            tf.Assert(tf.equal(tf.rank(grid), 4), ["Grid must be 4D Tensor"]),
```
Comment (fsx950223) [inline tensorflow_addons/image/dense_image_warp.py:69]: tf.debugging.assert_equal?

File: tensorflow_addons/image/dense_image_warp.py [hunk 2]
```diff
@@ -46,33 +46,56 @@ def interpolate_bilinear(
...
+                tf.equal(query_shape[2], 2), ["Query points last dimension must be 2."]
+            ),
+        ]
+    ):
+        return _interpolate_bilinear_impl(grid, query_points, indexing, name)
 
-        # grid shape checks
-        grid_static_shape = grid.shape
+
+@tf.function
```
Comment (fsx950223) [inline tensorflow_addons/image/dense_image_warp.py:88]: Inner tf.function is necessary? There is a tf.function on interpolate_bilinear.

**Review by bhack — APPROVED** [review]

**Review by fsx950223 — APPROVED** [review]

**Pull request merged**